source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1279,12 +1301,15 @@ implementation::
                     if isNotMaster(error):
                         check failing server
             elif isNetworkError(error) or (not error.completedHandshake and (isNetworkTimeout(error) or isAuthError(error))):
-                # Mark the server Unknown
-                unknown = new ServerDescription(type=Unknown, error=error)
-                onServerDescriptionChanged(unknown, connection pool for server)
-                clear connection pool for server
-                # Cancel inprogress check
-                cancel monitor check
+                if type != LoadBalanced
+                  # Mark the server Unknown
+                  unknown = new ServerDescription(type=Unknown, error=error)
+                  onServerDescriptionChanged(unknown, connection pool for server)
+                  clear connection pool for server
+                  # Cancel inprogress check
```

Comment:
We want to do this for LB mode as well. Can we move it outside the if/else so it's done unconditionally?

Reply:
Moved out to cover all cases.